src/Psy/Configuration.php

```diff
@@ -1110,7 +1110,12 @@ public function getChecker()
                 case Checker::DAILY:
                 case Checker::WEEKLY:
                 case Checker::MONTHLY:
-                    $this->checker = new IntervalChecker($this->getUpdateCheckCacheFile(), $interval);
+                    $check_file = $this->getUpdateCheckCacheFile();
+                    if ($check_file === false) {
+                        $this->checker = new NoopChecker();
```

Comment:
i think falling back to a `NoopChecker` when the user hasn't explicitly turned off update checking is probably grounds for showing an error message or warning of some sort.